example-trailhead/src/main/java/com/xjeffrose/xio/server/trailhead/Trailhead.java

```diff
@@ -26,22 +20,22 @@
   private final ApplicationBootstrap bootstrap;
   private Application application;
 
-  private RouteProvider buildProvider(ProxyConfig proxyConfig) {
-    return new SimpleProxyRoute(proxyConfig);
+  private RequestHandler buildProvider(Route route, ProxyConfig proxyConfig) {
```

Comment:
I waffled on calling this RequestHandler in the past. I don't have a good rubric for how we should name these things, we should probably spend some time clarifying concepts post PoC.

Reply:
i saw the todo and thought requesthandler made more sense to me so i went for it...  not totally married to it or anything though.